src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php

```diff
@@ -108,7 +108,7 @@ public function getProxyCode(Definition $definition)
      */
     private function getProxyClassName(Definition $definition)
     {
-        return preg_replace('/^.*\\\\/', '', $definition->getClass()).'_'.substr(hash('sha256', spl_object_hash($definition).$this->salt), -7);
+        return preg_replace('/^.*\\\\/', '', $definition->getClass()).'_'.substr(hash('sha256', serialize($definition).$this->salt), -7);
```

Comment:
I just looked at the current logic. It looks like this would be enough isn't it? (with lazy prefix to easy recognizing them)
`return 'lazy'.preg_replace('/^.*\\\\/', '', $definition->getClass()).'_'.substr(hash('sha256', $definition->getClass()), -7);`

Reply:
Indeed, that should be enough. The only scenario where this could break is if we would have the same class name with a different shape in one process. That is probably achievable through runkit and the like but I guess that’s a case we can ignore.